core/src/components/modal/modal.tsx

```diff
@@ -689,6 +689,26 @@ export class Modal implements ComponentInterface, OverlayInterface {
     return this.currentBreakpoint;
   }
 
+  private onHandleClick = () => {
+    const { breakpoints, currentBreakpoint } = this;
+    if (breakpoints && breakpoints.length > 1 && currentBreakpoint !== undefined) {
+      const currentBreakpointIndex = breakpoints.indexOf(currentBreakpoint);
+      /**
+       * Defaults to the first breakpoint. If the first breakpoint is 0, we select
+       * the second breakpoint to avoid tapping the handle to dismiss the sheet.
+       */
+      let nextBreakpoint = breakpoints[0] || breakpoints[1];
+      if (currentBreakpointIndex !== breakpoints.length - 1) {
+        /**
+         * Since the current breakpoint is not the last breakpoint,
+         * we advance to the next breakpoint.
+         */
+        nextBreakpoint = breakpoints[currentBreakpointIndex + 1];
```

Comment:
Optional, An alternative way of doing this:

```suggestion

     const allowedBreakpoints = breakpoints.filter(b => b !== 0);
     const currentBreakpointIndex = allowedBreakpoints.indexOf(currentBreakpoint);
     const nextBreakpoint = (currentBreakpointIndex + 1) % allowedBreakpoints.length;
     
     this.setCurrentBreakpoint(nextBreakpoint);
```

The modulo operator ensures we never go out of bounds on the breakpoints array.

Reply:
This solution will need to be adapted to:
```ts
const allowedBreakpoints = breakpoints.filter(b => b !== 0);
const currentBreakpointIndex = allowedBreakpoints.indexOf(currentBreakpoint);
const nextBreakpointIndex = (currentBreakpointIndex + 1) % allowedBreakpoints.length + 1;

this.setCurrentBreakpoint(breakpoints[nextBreakpointIndex]);
```

Plugging the original in; `nextBreakpoint` was actually the index position of the breakpoint. After swapping that to `breakpoints[nextBreakpoint]`, it was always evaluating the current breakpoint instead of the next value.

Does the updated snippet look correct to you? Confirmed locally that it loops through the values as expected.

Reply:
That works as well. Another option is to just use `allowedBreakpoints` when calling `setCurrentBreakpoint`. That way you don't have to add `1` to the result every time.